client/src/redux/app-mount-saga.js

```diff
@@ -2,23 +2,30 @@ import qs from 'query-string';
 import { put, takeEvery } from 'redux-saga/effects';
 import { createFlashMessage } from '../components/Flash/redux';
 
+function* clearSearchWithoutReRender() {
+  // Clear the search without causing a re-render
+  const newUrl = window.location.pathname;
+  window.history.pushState({}, '', newUrl);
+}
+
 function* parseMessagesSaga() {
   const search = window.location.search.slice();
-  // TODO(Bouncey): Find a way to clear the search with causing a re-render
   if (search) {
     const { messages } = qs.parse(search, { arrayFormat: 'index' });
     if (messages) {
       const flashMap = qs.parse(messages, { arrayFormat: 'index' });
       const flash = Object.keys(flashMap).reduce(
         (acc, type) => [
           ...acc,
-          ...flashMap[type].map(message => ({ type, message }))
+          ...flashMap[type].map(message => ({ type, message })),
         ],
         []
       );
       for (let i = 0; i < flash.length; i++) {
         yield put(createFlashMessage(flash[i]));
       }
+      // After processing the flash messages, clear the search parameters without re-render
+      yield clearSearchWithoutReRender();
```

Comment:
```suggestion
      clearSearchWithoutReRender();
```
There's nothing to wait for, so there's no need to yield.